lib/puppet/provider/truststore_certificate/truststore_certificate.rb

```diff
@@ -0,0 +1,77 @@
+Puppet::Type.type(:truststore_certificate).provide(:truststore_certificate) do
+
+  commands :keytool => 'keytool'
+  commands :openssl => 'openssl'
+
+  def create
+    add_certificate unless certificate_present?
+    if certificate_changed?
+      delete_certificate
+      add_certificate
+    end
+  end
+
+  def exists?
+    truststore_exists? && !certificate_changed?
+  end
```

Comment:
I think it would be better to override `insync?` on the property. https://github.com/theforeman/puppet-foreman/blob/c3c090ce8645a3f3da9cf7290ac13cab1503d8c6/lib/puppet/type/foreman_smartproxy.rb#L28-L43 is an example of how you can compare the certificate.

I think this part is what actually implements the property: https://github.com/theforeman/puppet-foreman/blob/c3c090ce8645a3f3da9cf7290ac13cab1503d8c6/lib/puppet/provider/foreman_smartproxy/rest_v3.rb#L43-L60

So `truststore_contents` would actually be the property `certificate` (in the provider) and then in the type you would compare those 2. `certificate=` (in the provider) would call `add_certificate` and `delete_certificate`.

Reply:
I opted to update ensurable's `insync` to decide if something should be changed. I found tacking this onto a single property to be confusing when it's the entirety of the resource we care about.